test/parallel/test-stream2-writable.js

```diff
@@ -402,3 +402,21 @@ const helloWorldBuffer = Buffer.from('hello world');
   w.write(Buffer.allocUnsafe(1));
   w.end(Buffer.allocUnsafe(0));
 }
+
+{
+  // Verify that finish is not emitted after error
+  const w = new W();
+
+  w._final = common.mustCall(function(cb) {
+    cb(new Error());
+  });
+  w._write = function(chunk, e, cb) {
+    process.nextTick(cb);
+  };
+  w
+    .on('error', common.mustCall())
```

Comment:
style nit: can we keep this and the previous line together. The `w` on a line by itself is rather odd